### PR TITLE
Added history of active clusters to NamespaceDetail

### PR DESCRIPTION
**Pull request by roselander**
**Note: ** This is a re-submit of https://github.com/temporalio/temporal/pull/2886 which was approved, closed, and cannot be re-opened.

What changed?
UpdateNamespace now checks whether the update results in a new cluster being active and if so records the time it became active in the NamespaceDetail.

How did you test it?
Existing/new unit tests.

Is hotfix candidate?
No

### CODE REVIEW

File: common/namespace/handler.go [hunk 1]
```diff
@@ -31,6 +31,8 @@ import (
 	"fmt"
 	"time"
 
+	"go.temporal.io/server/common/clock"
```
Comment (yux0) [inline common/namespace/handler.go:34]: nit: group with other go.temporal.io/server imports

File: common/namespace/handler.go [hunk 2]
```diff
@@ -795,6 +815,52 @@ func (d *HandlerImpl) validateVisibilityArchivalURI(URIString string) error {
...
+		now := d.timeSource.Now()
+		failoverHistory = append(
+			failoverHistory, &persistencespb.FailoverStatus{
+				FailoverTime:    &now,
+				FailoverVersion: newFailoverVersion,
+			},
+		)
+	}
+	if l := len(failoverHistory); l > maxReplicationHistorySize {
+		failoverHistory = failoverHistory[l-maxReplicationHistorySize : l]
```
Comment (yux0) [inline common/namespace/handler.go:859]: is this the same as failoverHistory[1:]?

File: common/namespace/handler.go [hunk 3]
```diff
@@ -550,18 +560,28 @@ func (d *HandlerImpl) UpdateNamespace(
...
 				failoverVersion,
 			)
 			failoverNotificationVersion = notificationVersion
 		}
+		// set the versions
+		if configurationChanged {
+			configVersion++
+		}
+
+		if configurationChanged || activeClusterChanged {
```
Comment (yux0) [inline common/namespace/handler.go:575]: if activeClusterChanged?

File: common/namespace/handler.go (hunk 1 above)
Comment (roselander) [reply common/namespace/handler.go]: Done

File: common/namespace/handler.go (hunk 2 above)
Comment (roselander) [reply common/namespace/handler.go:858]: No, it is taking the last `maxReplicationHistorySize` elements.

**Review by yux0 — APPROVED** [review]

File: common/namespace/handler.go (hunk 3 above)
Comment (roselander) [reply common/namespace/handler.go:574]: Transcribing offline discussion: Left check in place for safety, but added comment.

**Pull request merged**